### ISSUE TITLE
HIP and METAL compile error: bracket nesting level exceeded maximum of 256

### ISSUE THREAD

**Issue by flammit** [tinygrad/tinygrad#3328]
The error message is:

```
/tmp/comgr-ecf581/input/<null>:577:6263: fatal error: bracket nesting level exceeded maximum of 256
/tmp/comgr-ecf581/input/<null>:577:6263: note: use -fbracket-depth=N to increase maximum nesting level
1 error generated when compiling for gfx1100.
```

To reproduce use this Linearizer:

AST:
```
LazyOp(op=BufferOps.STORE, src=(LazyOp(op=ReduceOps.SUM, src=(LazyOp(op=BinaryOps.MUL, src=(LazyOp(op=BufferOps.LOAD, src=(), arg=MemBuffer(idx=1, dtype=dtypes.half, st=ShapeTracker(views=(View(shape=(1, 512, 1, 64, 4, 18, 4, 18), strides=(0, 16384, 0, 256, 0, 16, 0, 1), offset=-17, mask=((0, 1), (0, 512), (0, 1), (0, 64), (0, 4), (1, 17), (0, 4), (1, 17)), contiguous=False), View(shape=(512, 1, 64, 16, 16, 64, 3, 3), strides=(331776, 0, 0, 72, 1, 5184, 1368, 19), offset=0, mask=None, contiguous=False))))), LazyOp(op=UnaryOps.CAST, src=(LazyOp(op=BufferOps.LOAD, src=(), arg=MemBuffer(idx=2, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(512, 1, 64, 16, 16, 64, 3, 3), strides=(16384, 0, 256, 16, 1, 0, 0, 0), offset=0, mask=None, contiguous=False),)))),), arg=(dtypes.half, False))), arg=None),), arg=(1, 1, 64, 1, 1, 64, 3, 3)),), arg=MemBuffer(idx=0, dtype=dtypes.half, st=ShapeTracker(views=(View(shape=(1, 1, 64, 1, 1, 64, 3, 3), strides=(0, 0, 576, 0, 0, 9, 3, 1), offset=0, mask=None, contiguous=True),))))
```

APPLIED_OPTS:
```
[Opt(op=OptOps.LOCAL, axis=0, amt=32), Opt(op=OptOps.UNROLL, axis=2, amt=0), Opt(op=OptOps.UNROLL, axis=1, amt=0)]
```

**Comment by flammit** [comment]
Actually METAL has the same issue:

```
AssertionError: Error Domain=MTLLibraryErrorDomain Code=3 "program_source:79:5786: fatal error: bracket nesting level exceeded maximum of 256
```

**Comment by flammit** [comment]
Closing as invalid examples after reduce_op axis change in #3462

**Comment by geohot** [comment]
This is still a bug AFAIK

**Comment by flammit** [comment]
sorry I didn't mean to close this impliying that it was fixed.  when the reduce-axis changes were on master, this example wasn't valid anymore so I closed it and added one that was relevant in #3495

**Comment by chenyuxyz** [comment]
fixed by #3634 and #3635. Feel free to open a new one with new ast
